cpp/src/io/fst/agent_dfa.cuh

```diff
@@ -91,13 +91,24 @@ class DFASimulationCallbackWrapper {
                                                       SymbolT const read_symbol)
   {
     uint32_t const count = transducer_table(old_state, symbol_id, read_symbol);
+#if __CUDA_ARCH__ > 0
```

Comment:
This will disable loop unroll for only the `device` code? 
This duplicate code path seems messy and borderline [ODR](https://en.cppreference.com/w/cpp/language/definition).
Would it be ok to disable it for the `host` path as well?

Reply:
When using `#pragma unroll` within the host compilation pass, I'm getting an `error: ignoring '#pragma unroll ' [-Werror=unknown-pragmas]`. What would be your suggested solution to this?

Reply:
It looks like the gcc equivalent is `#pragma GCC unroll 1`
Perhaps the for-loop could be wrapped with
```
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wunknown-pragmas"
#pragma unroll 1
...(for-loop goes here)...
#pragma GCC diagnostic pop
```

Reply:
Ok, none of these worked for me. Does this work?
```
#if __CUDA_ARCH__ > 0
#pragma unroll 1
#endif
```

Reply:
will this suggestion work?
```
#if __CUDA_ARCH__ > 0
      #pragma unroll 1
      #endif
```

Reply:
Thank you for your suggestions. I've adopted that.